HealthcareApp/src/Healthcare.Application/Appointments/Commands/Book/BookAppointmentCommandHandler.cs

```diff
@@ -0,0 +1,60 @@
+using Application.Abstractions;
+using Application.Abstractions.Decorators;
+using Domain.Entities;
+using Healthcare.Application.Appointments.Commands.Book;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using static Domain.Constants.AppointmentStatusConstants;
+
+namespace Healthcare.Application.Appointments.Commands;
+
+public class BookAppointmentCommandHandler : IRequestHandler<BookAppointmentCommand, bool>
+{
+    private readonly IUserManagerDecorator<ApplicationUser> _userManager;
+    private readonly IAppointmentRepository _appointmentRepository;
+
+
+    public BookAppointmentCommandHandler(IUserManagerDecorator<ApplicationUser> userManager,
+        IAppointmentRepository appointmentRepository)
+    {
+        _userManager = userManager;
+        _appointmentRepository = appointmentRepository;
+    }
+
+    public async Task<bool> Handle(BookAppointmentCommand request, CancellationToken cancellationToken)
+    {
+        var users = await _userManager.GetAllUsersAsync();
```

Comment:
Your application will suffer from performance issues as soon as your user table gets larger.
Think about optimization - for example you might query a single user for this use case